### PR TITLE
added support for symfony 6.2

### PR DESCRIPTION
**Pull request by marinhekman**
you can switch to Symfony 6.2 by calling "bash install.sh 6.2"
or back to the default 5.4 by calling "bash install.sh"

so to explain, symfony configuration requires these 4 things:
1. folder config/
2. file composer.json
3. file composer.lock
4. file symfony.lock

the install.sh script just creates symlinks to:
1. config => config-symfonyX.Y
2. composer.json => composer-symfonyX.Y.json
3. composer.lock => composer-symfonyX.Y.lock
4. symfony.lock => symfonyX.Y.json

where X.Y can be 5.4 or 6.2

### CODE REVIEW

File: ServerManager/init.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
 require_once 'ServerManager/helpers.php';
 
 // let's get going!
-session_start();
+session_id() or session_start();
```
Comment (marinhekman) [inline ServerManager/init.php:17]: this fixes a notice error when calling "start" if the session is already started

File: bin/phpunit [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env php
```
Comment (marinhekman) [inline bin/phpunit:1]: copied this file from an empty symfony 6.2 checkout

File: composer-symfony5.4.json [hunk 3]
```diff
@@ -115,7 +115,7 @@
         }
     },
     "require-dev": {
-        "brainmaestro/composer-git-hooks": "dev-master",
+        "brainmaestro/composer-git-hooks": "dev-master#a714e58968a37dba2783417a5d26ef4b1afd8268",
```
Comment (marinhekman) [inline composer-symfony5.4.json:118]: refering to old commit such that there are no changes for 5.4

File: composer-symfony6.2.json [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                "vendor/bin/phpstan analyse"
+            ]
+        },
+        "symfony": {
+            "allow-contrib": false,
+            "require": "6.2.*"
+        }
+    },
+    "require-dev": {
+        "brainmaestro/composer-git-hooks": "dev-master",
```
Comment (marinhekman) [inline composer-symfony6.2.json:141]: note that the master has been updated to the latest version to support symfony 6.2

File: config-symfony6.2/ServerManager/routes.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+server_manager_index:
```
Comment (marinhekman) [inline config-symfony6.2/ServerManager/routes.yaml:1]: identical to 5.4

File: config-symfony6.2/packages/doctrine.yaml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+imports:
```
Comment (marinhekman) [inline config-symfony6.2/packages/doctrine.yaml:1]: identical to 5.4

File: config-symfony6.2/packages/doctrine.php [hunk 7]
```diff
@@ -0,0 +1,43 @@
+<?php
```
Comment (marinhekman) [inline config-symfony6.2/packages/doctrine.php:1]: almost identical to 5.4

File: config-symfony6.2/packages/doctrine_migrations.yaml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+doctrine_migrations:
```
Comment (marinhekman) [inline config-symfony6.2/packages/doctrine_migrations.yaml:1]: identical to 5.4

File: config-symfony6.2/packages/framework.yaml [hunk 9]
```diff
@@ -0,0 +1,34 @@
+# see https://symfony.com/doc/current/reference/configuration/framework.html
```
Comment (marinhekman) [inline config-symfony6.2/packages/framework.yaml:1]: identical to 5.4

File: config-symfony6.2/packages/monolog.yaml [hunk 10]
```diff
@@ -0,0 +1,61 @@
+monolog:
```
Comment (marinhekman) [inline config-symfony6.2/packages/monolog.yaml:1]: identical to 5.4

File: config-symfony6.2/packages/prod/monolog.yaml [hunk 11]
```diff
@@ -0,0 +1,18 @@
+# config/packages/prod/monolog.yaml
```
Comment (marinhekman) [inline config-symfony6.2/packages/prod/monolog.yaml:1]: identical to 5.4

File: config-symfony6.2/routes.yaml [hunk 12]
```diff
@@ -0,0 +1,65 @@
+server_manager:
```
Comment (marinhekman) [inline config-symfony6.2/routes.yaml:1]: identical to 5.4

File: config-symfony6.2/routes/tools.yaml [hunk 13]
```diff
@@ -0,0 +1,8 @@
+when@dev:
```
Comment (marinhekman) [inline config-symfony6.2/routes/tools.yaml:1]: identical to 5.4

File: config-symfony6.2/services.yaml [hunk 14]
```diff
@@ -0,0 +1,57 @@
+# This file is the entry point to configure your own services.
```
Comment (marinhekman) [inline config-symfony6.2/services.yaml:1]: identical to 5.4

File: composer-symfony6.2.json [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+        "symfony/process": "6.2.*",
+        "symfony/property-access": "6.2.*",
+        "symfony/property-info": "6.2.*",
+        "symfony/runtime": "6.2.*",
+        "symfony/security-bundle": "6.2.*",
+        "symfony/serializer": "6.2.*",
+        "symfony/string": "6.2.*",
+        "symfony/translation": "6.2.*",
+        "symfony/twig-bundle": "6.2.*",
+        "symfony/uid": "6.2.*",
```
Comment (marinhekman) [inline composer-symfony6.2.json:55]: had to require this separately, I think is not part of Symfony base anymore but a separate lib

File: config-symfony6.2/bundles.php [hunk 16]
```diff
@@ -0,0 +1,16 @@
+<?php
```
Comment (marinhekman) [inline config-symfony6.2/bundles.php:1]: Different from 5.4, they restructurized the libs it seems

File: config-symfony6.2/packages/cache.yaml [hunk 17]
```diff
@@ -0,0 +1,19 @@
+framework:
```
Comment (marinhekman) [inline config-symfony6.2/packages/cache.yaml:1]: identical to 5.4

File: config-symfony6.2/packages/debug.yaml [hunk 18]
```diff
@@ -0,0 +1,5 @@
+when@dev:
```
Comment (marinhekman) [inline config-symfony6.2/packages/debug.yaml:1]: new to 6.2

File: config-symfony6.2/packages/mailer.yaml [hunk 19]
```diff
@@ -0,0 +1,3 @@
+framework:
```
Comment (marinhekman) [inline config-symfony6.2/packages/mailer.yaml:1]: new to 6.2

File: config-symfony6.2/packages/messenger.yaml [hunk 20]
```diff
@@ -0,0 +1,24 @@
+framework:
```
Comment (marinhekman) [inline config-symfony6.2/packages/messenger.yaml:1]: new to 6.2

File: config-symfony6.2/packages/routing.yaml [hunk 21]
```diff
@@ -0,0 +1,12 @@
+framework:
```
Comment (marinhekman) [inline config-symfony6.2/packages/routing.yaml:1]: identical to 5.4

File: config-symfony6.2/packages/security.yaml [hunk 22]
```diff
@@ -0,0 +1,39 @@
+security:
```
Comment (marinhekman) [inline config-symfony6.2/packages/security.yaml:1]: new to 6.2

File: config-symfony6.2/packages/translation.yaml [hunk 23]
```diff
@@ -0,0 +1,13 @@
+framework:
```
Comment (marinhekman) [inline config-symfony6.2/packages/translation.yaml:1]: identical to 5.4

File: config-symfony6.2/packages/twig.yaml [hunk 24]
```diff
@@ -0,0 +1,6 @@
+twig:
```
Comment (marinhekman) [inline config-symfony6.2/packages/twig.yaml:1]: identical to 5.4

File: config-symfony6.2/packages/validator.yaml [hunk 25]
```diff
@@ -0,0 +1,13 @@
+framework:
```
Comment (marinhekman) [inline config-symfony6.2/packages/validator.yaml:1]: new to 6.2

File: config-symfony6.2/packages/web_profiler.yaml [hunk 26]
```diff
@@ -0,0 +1,17 @@
+when@dev:
+    web_profiler:
+        toolbar: true
+        intercept_redirects: false
+
+    framework:
+        profiler:
+            only_exceptions: false
+            collect_serializer_data: true
```
Comment (marinhekman) [inline config-symfony6.2/packages/web_profiler.yaml:9]: identical to 5.4 except for this new line added

File: config-symfony6.2/routes/framework.yaml [hunk 27]
```diff
@@ -0,0 +1,4 @@
+when@dev:
```
Comment (marinhekman) [inline config-symfony6.2/routes/framework.yaml:1]: identical to 5.4

File: set_symfony_version.sh [hunk 28]
```diff
@@ -2,19 +2,19 @@
...
-VERSION_DEFAULT="5.4"
+SYMFONY_VERSION_DEFAULT="5.4"
 
 setSymfonyVersion() {
-  VERSION=$1
-  if [[ -z $1 || "$1" == "default" ]]; then
-    VERSION="$VERSION_DEFAULT"
+  [[ -z "${SYMFONY_VERSION}" ]] && SYMFONY_VERSION=$1
+  if [[ -z $SYMFONY_VERSION || "${SYMFONY_VERSION}" == "default" ]]; then
+    SYMFONY_VERSION="${SYMFONY_VERSION_DEFAULT}"
```
Comment (marinhekman) [inline set_symfony_version.sh:10]: added support for environmental variable SYMFONY_VERSION
such that you can run:

export SYMFONY_VERSION=6.2

and then call "bash install.sh" 
so no need to add "6.2" as an argument everytime if you want to switch locally

**Review by JacopoFabrini — APPROVED** [review]
Thanks for all the comments/explanations!

**Pull request merged**